service/controller/v14/credential/credential.go

```diff
@@ -28,6 +28,24 @@ func GetARN(k8sClient kubernetes.Interface, obj interface{}) (string, error) {
 	return arn, nil
 }
 
+// GetDefaultARN is used only by the bridgezone resource. It should be removed
+// when the resource is removed.
+func GetDefaultARN(k8sClient kubernetes.Interface) (string, error) {
```

Comment:
Any reason why there is no service like implementation which is configured with a k8s client? When this is for BYOC I thought we have the clients being transported via the context so I would have expected this mechanism to be used here.

Reply:
I just followed the pattern we already have https://github.com/giantswarm/aws-operator/pull/1060/files#diff-2a88f9bcdf0b0ee008cb4eae77ddc9b5R17

This function is here only for the migration period. As comment says it should be removed when `bridgezone` resource is removed.